### ISSUE TITLE
Override rulefiles

### ISSUE THREAD

**Issue by janotav** [prometheus-operator/prometheus-operator#3468]
When prometheus-operator creates the final configuration for prometheus it puts prometheusrules into paths looking like:

/etc/prometheus/rules/prometheus-release-operator-prometheus-rulefiles-0/<**rule_metadata_namespace>-<rule_metadata_name>.yaml**

Is it possible to override this pattern? Can you please point me to where the relevant code resides?

My use-case is probably very specific and might be outside the prometheus-operator scope and I might have to resort to my own implementation, but I would like to know what are the options here.

Thanks.

**Comment by janotav** [comment]
After some more digging I discovered that my use-case is kind of generalization of _prometheus.prometheusSpec.enforcedNamespaceLabel_

This can be used to "tag" the objects (including rules) with a label that will receive the name of the enclosing namespace.

In my scenario each namespace has its owner, but single owner can own multiple namespaces. The owner of a namespace is defined by a well-known label on the namespace:

```
kind: Namespace
metadata:
  name: namespace-name
  labels:
    owner: team-name
```
What I would like to do is to enforce the value of the well-known label (called _owner_ in this case) that represents ownership rather than the simple namespace name.

This would in turn allow me to route alerts from all namespaces to its owning team.

**Comment by janotav** [comment]
I managed to to come to a reasonable solution for my scenario.

1. I patched prometheus-operator rule filename pattern to use "--" instead of "-" 
2. this allows me to reliably detect namespace from the rule_group label using regex (because in our case namespace name may contain "-" but must not contain "--")
3. this in turn allows me to "join" on the prometheusrule namespace in recording rule
4. finally alert is created on top of this recorded rule

As a result the owner of the namespace who owns the failing prometheusrule receives the alert. Small caveat is that due to the rule indirection, the alert takes a little longer to resolve, but that is negligible.

I am closing this because I assume that this use-case is probably too specific and the wider community would probably not benefit from the ability to override the delimiter in the rule filenames.
